src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -691,6 +772,16 @@ public void SetAnalogPinSamplingInterval(TimeSpan timeSpan)
             Firmata.SetAnalogInputSamplingInterval(timeSpan);
         }
 
+        /// <summary>
+        /// Performs a software reset of the Arduino firmware
+        /// </summary>
+        public void SoftwareReset()
+        {
+            Initialize();
+            Firmata.SendSoftwareReset();
```

Comment:
Should all these APIs be checking if firmata is disposed?

Reply:
The call to `Initialize()` performs that test.